packages/shared/src/hooks/useOrganizationList.tsx

```diff
@@ -1,39 +1,248 @@
+  data: T[];
+  count: number;
+  isLoading: boolean;
+  isFetching: boolean;
+  isError: boolean;
+  page: number;
+  pageCount: number;
+  fetchPage: CustomSetAction<number>;
+  fetchPrevious: () => void;
+  fetchNext: () => void;
+  hasNextPage: boolean;
+  hasPreviousPage: boolean;
+};
+
+// Utility type to convert PaginatedDataAPI to properties as undefined, except booleans set to false
+type PaginatedDataAPIWithDefaults<T> = {
+  [K in keyof PaginatedDataAPI<T>]: PaginatedDataAPI<T>[K] extends boolean ? false : PaginatedDataAPI<T>[K] | undefined;
+};
+
 type UseOrganizationListReturn =
-  | { isLoaded: false; organizationList: undefined; createOrganization: undefined; setActive: undefined }
   | {
-      isLoaded: true;
+      isLoaded: false;
+      organizationList: undefined;
+      createOrganization: undefined;
+      setActive: undefined;
+      userInvitations: PaginatedDataAPIWithDefaults<UserOrganizationInvitationResource>;
+    }
+  | {
+      isLoaded: boolean;
       organizationList: OrganizationList;
       createOrganization: (params: CreateOrganizationParams) => Promise<OrganizationResource>;
       setActive: SetActive;
+      userInvitations: PaginatedDataAPI<UserOrganizationInvitationResource>;
     };
 
-type UseOrganizationList = () => UseOrganizationListReturn;
+type UseOrganizationList = (params?: UseOrganizationListParams) => UseOrganizationListReturn;
+
+export const useOrganizationList: UseOrganizationList = params => {
+  const { userInvitations } = params || {};
+
+  const shouldUseDefaults = typeof userInvitations === 'boolean' && userInvitations;
+  const [paginatedPage, setPaginatedPage] = useState(shouldUseDefaults ? 1 : userInvitations?.initialPage ?? 1);
+
+  // Cache initialPage and initialPageSize until unmount
+  const initialPageRef = useRef(shouldUseDefaults ? 1 : userInvitations?.initialPage ?? 1);
+  const initialPageSizeRef = useRef(shouldUseDefaults ? 10 : userInvitations?.initialPageSize ?? 10);
+
+  const triggerInfinite = shouldUseDefaults ? false : !!userInvitations?.infinite;
+  const internalKeepPreviousData = shouldUseDefaults ? false : !!userInvitations?.keepPreviousData;
+  const internalStatus = shouldUseDefaults ? 'pending' : userInvitations?.status ?? 'pending';
 
-export const useOrganizationList: UseOrganizationList = () => {
   const clerk = useClerkInstanceContext();
   const user = useUserContext();
 
+  const paginatedParams =
+    typeof userInvitations === 'undefined'
+      ? undefined
+      : {
+          initialPage: paginatedPage,
+          initialPageSize: initialPageSizeRef.current,
+          status: internalStatus,
+        };
+
+  const canFetch = !!(clerk.loaded && user);
+
+  // Some gymnastics to adhere to the rules of hooks
+  // We need to make sure useSWR is called on every render
+  const fetchInvitations = !clerk.loaded
+    ? () => ({ data: [], total_count: 0 } as ClerkPaginatedResponse<UserOrganizationInvitationResource>)
```

Comment:
❓ Have we discussed the hook's return value before Clerk is ready? Is it possible to return `null` at this point as it's a bit of a code smell that we hardcode the format of the FAPI response in the hook? If not mistaken `null` is our go-to thing in other hooks for now, right?